### PR TITLE
Add stream orientation to camera entity registry settings

### PR DESCRIPTION
**Pull request by uvjustin**
## Proposed change

https://github.com/home-assistant/core/pull/77439 adds the ability to apply orientation transforms in stream.
This PR allows the frontend to change those settings by adding a selector for the stream orientation to the camera entity registry settings.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by uvjustin** [comment]
Core PR was merged last week

File: src/panels/config/entities/entity-registry-settings.ts [hunk 1]
```diff
@@ -595,6 +599,38 @@ export class EntityRegistrySettings extends SubscribeMixin(LitElement) {
...
+                <ha-select
+                  .label=${this.hass.localize(
+                    "ui.dialogs.entity_registry.editor.stream.stream_orientation"
+                  )}
+                  naturalMenuWidth
+                  fixedMenuPosition
+                  @selected=${this._handleCameraOrientationChanged}
+                  @closed=${stopPropagation}
+                >
+                  ${[1, 2, 3, 4, 6, 8].map((num) => {
```
Comment (bramkragten) [inline src/panels/config/entities/entity-registry-settings.ts:622]: Make `[1, 2, 3, 4, 6, 8]` a const in `data/camera.ts`

File: src/panels/config/entities/entity-registry-settings.ts [hunk 2]
```diff
@@ -836,6 +872,21 @@ export class EntityRegistrySettings extends SubscribeMixin(LitElement) {
     }
   }
 
+  private async _handleCameraOrientationChanged(ev) {
+    const selector = ev.currentTarget as HaSelector;
```
Comment (bramkragten) [inline src/panels/config/entities/entity-registry-settings.ts:876]: It is a `HaSelect` not `HaSelector`

File: src/data/camera.ts [hunk 3]
```diff
@@ -26,6 +27,7 @@ export interface CameraEntity extends HassEntityBase {
 
 export interface CameraPreferences {
   preload_stream: boolean;
+  orientation: number;
```
Comment (bramkragten) [inline src/data/camera.ts:30]:
```ts
type ValueOf<T extends any[]> = T[number];

orientation: ValueOf<typeof CAMERA_ORIENTATIONS>;
```

Comment (bramkragten) [reply src/data/camera.ts:30]: here is a nice article: https://dev.to/smeijer/a-typescript-valueof-implementation-and-how-it-s-built-4gim

**Review by bramkragten — APPROVED** [review]

**Pull request merged**